### PR TITLE
Feature/teacher

### LINKED ISSUE #19: As a Teacher, I want to add my students to my roster

### LINKED ISSUE #48: Write a thunk POST request to add/remove a student
- [ ] Write a thunk POST request to add/remove a student

### LINKED ISSUE #49: - [ ] Write a thunk POST request to add a lesson to the database
- [ ] Write a thunk POST request to add a lesson to the database

### PR DESCRIPTION
**Pull request by KyleWong2510**
## Pull request checklist
Please check if your PR fulfills the following requirements:
- [ ] Lint (`npm run lint`) has passed locally and any fixes were made for failures
## Pull request type
 
 
Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Testing
- [ ] Other (please describe): 
## What is the current behavior?

- No modals
- No POSTing

## What is the new behavior?

- Added Modal, CreateStudent, and StudentDetails components
- POST functionality for students and lessons
- Fix POST issues with BE help

## Does this introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

closes #19
closes #48
closes #49

### CODE REVIEW

File: src/CreateStudentForm/CreateStudentForm.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import React, { useState } from 'react'
```
Comment (Billwilke42) [inline src/CreateStudentForm/CreateStudentForm.js:1]: Nice work on this! Adds so much ability to the MVP

File: src/CreateStudentForm/CreateStudentForm.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      <section>
+        <label className='student-name-label' htmlFor='student-firstname-input'>First Name:</label>
+        <input
+          id='student-firstname-input' 
+          className='student-name-input'
+          type='text'
+          placeholder='Enter Student First Name...'
+          value={studentFirstName}
+          onChange={(e) => setStudentFirstName(e.target.value)}
+        />
```
Comment (Melizzo) [inline src/CreateStudentForm/CreateStudentForm.js:60]: This is great - making every student a click - totally stealing this for the lessons.

File: src/CreateStudentForm/CreateStudentForm.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        className='create-student-submit-btn'
+        type='submit'
+        value='Add Student'
+        disabled={isEnabled}
+      />
+    </form>
+  )
+}
+
+const mapDispatchToProps = (dispatch) =>
```
Comment (Billwilke42) [inline src/CreateStudentForm/CreateStudentForm.js:83]: Will you need to have the students available in the state here?

File: src/LessonPreview/LessonPreview.js [hunk 4]
```diff
@@ -13,19 +13,40 @@ const LessonPreview = ({
...
+        answers: question.answers
+      }
+    })
     const lesson = {
-      name: lessonTitleText,
-      questions: questions,
+      lesson: {
+        name: lessonTitleText,
+        questions: formattedQuestions,
+      }
```
Comment (Melizzo) [inline src/LessonPreview/LessonPreview.js:32]: Interesting - this got updated with the backend?

File: src/TeacherRoster/TeacherRoster.js [hunk 5]
```diff
@@ -1,13 +1,64 @@
...
+  const renderStudentNames = () => {
+    return students.map(student => {
+      return (
+        <p id={student.id} onClick={(e) => findStudent(e)}>{student.first_name} {student.last_name}</p>
+      )
+    })
+  }
+
+  const findStudent = (e) => {
+    const found = students.find(student => +e.target.id === student.id)
```
Comment (Billwilke42) [inline src/TeacherRoster/TeacherRoster.js:22]: Whats this +e doing? I have never seen that before

File: src/LessonPreview/QuestionCard/QuestionCard.js [hunk 6]
```diff
@@ -1,11 +1,11 @@
 import React from 'react'
 import './QuestionCard.scss'
 
-const QuestionCard = ({ id, question, allAnswers, reading, deleteQuestion})=> {
-  let correctAnswer = allAnswers.find(answer => answer.correct === true).desc
+const QuestionCard = ({ id, question, allAnswers, reading, deleteQuestion })=> {
+  let correctAnswer = allAnswers.find(answer => answer.correct === 'true').answer
   let incorrectAnswers = allAnswers
```
Comment (Melizzo) [inline src/LessonPreview/QuestionCard/QuestionCard.js:6]: I see what you did here - great work Kyle.

File: src/actions/index.js [hunk 7]
```diff
@@ -8,6 +8,11 @@ export const setStudents = students => ({
   students
 })
 
+export const addStudent = student => ({
```
Comment (Billwilke42) [inline src/actions/index.js:11]: nice, this will be useful

File: src/Modal/Modal.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+import './Modal.scss'
+
+const Modal = ({ content, toggleDisplay }) => {
+  return (
+    <section className='modal'>
+      <button onClick={toggleDisplay}>X</button>
```
Comment (Melizzo) [inline src/Modal/Modal.js:7]: MODAL FTW!!

**Review by Melizzo — APPROVED** [review]

File: src/thunks/getStudents.js [hunk 9]
```diff
@@ -1,7 +1,7 @@
 import { isLoading, hasErrored, setStudents } from '../actions'
 
 export const getStudents = (teacherId) => {
-  const url = `https://gumberoo-backend.herokuapp.com/api/v1/teachers/${teacherId}/students`
+  const url = `https://gumberoo-backend.herokuapp.com/api/v1/teachers/1/students`
```
Comment (Billwilke42) [inline src/thunks/getStudents.js:4]: oh sorry I had to change this for my tests, but I will fix the tests so it passes

**Review by Billwilke42 — APPROVED** [review]
Nice work, we are so close to fulfilling the MVP. I should have the animations and completed student side up tonight. Might need more styling, but functionality will all be there. I was having some issues with the logic regarding disabling the buttons, maybe we could rubber duck it later?

File: src/LessonPreview/LessonPreview.js (hunk 4 above)
Comment (KyleWong2510) [reply src/LessonPreview/LessonPreview.js:32]: Yes this was the issue all along.

File: src/TeacherRoster/TeacherRoster.js (hunk 5 above)
Comment (Melizzo) [reply src/TeacherRoster/TeacherRoster.js:22]: Great Catch Bill - I haven't seen that either

File: src/CreateStudentForm/CreateStudentForm.js (hunk 3 above)
Comment (KyleWong2510) [reply src/CreateStudentForm/CreateStudentForm.js:83]: Probably not.  It is currently working on the redux store so it should be fine.

File: src/TeacherRoster/TeacherRoster.js (hunk 5 above)
Comment (KyleWong2510) [reply src/TeacherRoster/TeacherRoster.js:22]: Making it a number so we don't have to use the disable lint line

**Pull request merged**